### PR TITLE
feat: only expose https ports when https port is defined in the service

### PR DESCRIPTION
**Pull request by pyromaniac3010**
The https service and https containerports are typically not needed as most implementations will use the ingress with http backend communication. The https ports should therefore be disabled by default and not included in the service and statefulset. Only if someone manually sets a https port number via values, it should be exposed.
This also fixes some default prometheusrules which alert if service endpoints are not available.

### CODE REVIEW

**Comment by pyromaniac3010** [comment]
@grieshaber @mircohaug  Anything wrong with this PR?

**Comment by pyromaniac3010** [comment]
@grieshaber the mailhog chart was accidentally committed. this was caused by a rebase on my side. 
you could leave the default port setting as it was to avoid a major release. but I think it should be changed to null with the next release as ssl termination is usually done at the ingress level and not the container in kubernetes deployments.

File: charts/keycloakx/values.yaml [hunk 1]
```diff
@@ -236,7 +236,7 @@ service:
   # The HTTP Service node port if type is NodePort
   httpNodePort: null
   # The HTTPS Service port
-  httpsPort: 8443
+  httpsPort: null
```
Comment (grieshaber) [inline charts/keycloakx/values.yaml:239]: Could you please revert the changes to the defaults? I agree to disable the https as a default would make sense. But, i do not have the time to "fix" the release-automation, so in order to merge this change sooner, i'd suggest to stick to the default for now.

**Review by grieshaber — CHANGES_REQUESTED** [review]
It seems that the commit signature check is not succesful - https://github.com/codecentric/helm-charts/pull/651/checks?check_run_id=9886267385 could you have a look & commit with a valid signature?

File: charts/keycloakx/values.yaml (hunk 1 above)
Comment (pyromaniac3010) [reply charts/keycloakx/values.yaml]: done
